public/src/components/channelManagement/bannerTests/bannerUiSelector.tsx

```diff
@@ -0,0 +1,189 @@
+  { template: BannerTemplate.EnvironmentMomentBanner, label: 'Environment moment' },
+  { template: BannerTemplate.GlobalNewYearBanner, label: 'Global New Year moment' },
+  {
+    template: BannerTemplate.UkraineMomentBanner,
+    label: 'Ukraine Moment Banner 2023',
+  },
+  {
+    template: BannerTemplate.ChoiceCardsBannerBlue,
+    label: 'Choice cards banner - TABS',
+  },
+  {
+    template: BannerTemplate.ChoiceCardsButtonsBannerBlue,
+    label: 'Choice cards banner - BUTTONS',
+  },
+  {
+    template: BannerTemplate.WorldPressFreedomDayBanner,
+    label: 'World Press Freedom Day',
+  },
+  {
+    template: BannerTemplate.Scotus2023MomentBanner,
+    label: 'US Supreme Court 2023 Moment',
+  },
+];
+
+interface BannerTemplateSelectorProps {
+  template: BannerTemplate;
+  onUiChange: (updatedUi: BannerUi) => void;
+  editMode: boolean;
+}
+
+const BannerTemplateSelector: React.FC<BannerTemplateSelectorProps> = ({
+  template,
+  onUiChange,
+  editMode,
+}: BannerTemplateSelectorProps) => {
+  const onChange = (event: React.ChangeEvent<{ value: unknown }>): void => {
+    const value = event.target.value as string;
+    if (isBannerTemplate(value)) {
+      onUiChange(value);
+    }
+  };
+
+  return (
+    <Select value={template} onChange={onChange} disabled={!editMode}>
+      {templatesWithLabels.map(withLabel => (
+        <MenuItem value={withLabel.template} key={withLabel.template}>
+          {withLabel.label}
+        </MenuItem>
+      ))}
+    </Select>
+  );
+};
+
+interface BannerDesignSelectorProps {
+  design: BannerDesignName;
+  onUiChange: (updatedUi: BannerUi) => void;
+  editMode: boolean;
+  designs: BannerDesign[];
+}
+
+const BannerDesignSelector: React.FC<BannerDesignSelectorProps> = ({
+  design,
+  onUiChange,
+  editMode,
+  designs,
+}: BannerDesignSelectorProps) => {
+  const onChange = (event: React.ChangeEvent<{ value: unknown }>): void => {
+    const designName = event.target.value as string;
+    const isValidBannerDesign = designs.map(d => d.name).includes(designName);
+    if (isValidBannerDesign) {
+      onUiChange({ designName });
+    }
+  };
+
+  return (
+    <Select value={design.designName} onChange={onChange} disabled={!editMode}>
+      {designs.map(design => (
+        <MenuItem value={design.name} key={design.name}>
+          {design.name}
+        </MenuItem>
+      ))}
+    </Select>
+  );
+};
+
+type UiType = 'Design' | 'Template';
+
+const BannerUiSelector: React.FC<BannerUiSelectorProps> = ({
+  ui,
+  onUiChange,
+  editMode,
+  designs,
+}: BannerUiSelectorProps) => {
+  const [uiType, setUiType] = useState<UiType>(uiIsDesign(ui) ? 'Design' : 'Template');
+
+  const onUiTypeChange = (event: React.ChangeEvent<{ value: unknown }>): void => {
+    const uiType = event.target.value as UiType;
+    setUiType(uiType);
+
+    onUiChange(
+      uiType === 'Design' ? { designName: 'EXAMPLE' } : BannerTemplate.ContributionsBanner,
```

Comment:
Will this become something like `uiType === 'Design' ? designs[0] : ...`?
Shall we do this now?

Reply:
Or do we not want to initialise the design when the type changes?

Reply:
Great spot! This is an oversight (originally I hardcoded a list of designs, one of which was called `EXAMPLE` before wiring up fetching the real ones from Dynamo). It wasn't _obviously_ broken in the UI though because currently there really is one called `EXAMPLE` in the DEV dynamo table 😃

![Screenshot 2023-08-30 at 15 02 10](https://github.com/guardian/support-admin-console/assets/379839/96c479b5-4e60-464e-8579-1404419a104a)

I'll fix this now.